ts/components/ContextualHelpModal.tsx

```diff
@@ -43,101 +45,110 @@ const styles = StyleSheet.create({
 
 /**
  * A modal to show the contextual help reelated to a screen.
- * the contextual help is characterized by:
+ * The contextual help is characterized by:
  * - a title
  * - a textual or a component containing the screen description
  * - [optional] if on SPID authentication once the user selected an idp, content to link the support desk of the selected identity provider
  * - a list of questions and aswers. They are selected by the component depending on the cathegories passed to the component
+ *
+ * Optionally, the title and the content are injected from the content presented in the related clinet response.
  */
-export class ContextualHelpModal extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      content: null
-    };
-  }
+const ContextualHelpModal: React.FunctionComponent<Props> = (props: Props) => {
+  const [content, setContent] = React.useState<React.ReactNode | null>(null);
+
+  // after the modal is fully visible, render the content -
+  // in case of complex markdown this can take some time and we don't
+  // want to impact the modal animation
+  const onModalShow = () => setContent(props.body);
```

Comment:
```suggestion
  const onModalShow = () => setContent(props.body());
```

Reply:
updated. It is not clear why it does not return an error on the type (the body was rendered anyway)